articles/how-to-use-an-api-in-a-vuejs-project/index.md

```diff
@@ -0,0 +1,374 @@
+
+[Vuex](https://vuex.vuejs.org/) is a state management library for vue.js applications. It provides a centralized store for all elements in an application. 
+
+**Installing Vuex**
+
+To make use of VueX, you will first need to install the VueX package on your Vue application.
+
+```bash
+vue create project
+```
+
+OR
+
+```bash
+npm install vuex --save
+```
+
+Then, in your `store` folder, access the `index.js` file and write the following code
+
+```JavaScript
+import Vue from 'vue'
+import Vuex from 'vuex'
+
+Vue.use(Vuex);
+```
+
+You will also need to install and import Axios if you have not already done that. Check the earlier part of this tutorial for the steps.
+ 
+**Now let’s make the API request:**
+ 
+We will be working with the `store/index.js` file.
+First, we create a `state` object which will contain all the application-level state. It serves as the `data` object for the store in a vuex project.
+
+```JavaScript
+export default new Vuex.Store({
+ state: {
+    posts: [],
+  },
+
+}) 
+```
+ 
+Next, we create a `getters` property. Getters are like `computed` properties for stores. It is used to determine derived states based on store states. In this tutorial, we will use it to return posts in the state.
+ 
+```JavaScript
+getters: {
+    posts: state => {
+        return state.posts;
+    }
+},
+```
+
+Next, we create a `mutation` property. The mutation property is where we can change the state in the Vuex store. There are very similar to events where we carry out actual state alterations.
+
+```JavaScript
+mutations: {
+        SET_ITEMS (state, posts) {
+            state.posts = posts
+        }
+},
+```
+
+Now we can call our API in the `actions` property. Actions are equivalent to mutations only that actions commit mutations rather than mutating the state and also actions can hold asynchronous operations. Let’s go ahead with the API call.
+
+```JavaScript
+actions: {
+   async loadPosts ({ commit }) {
+     try {
+           const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+          // JSON responses are automatically parsed.
+          commit('SET_ITEMS', response.data)
+       }
+      catch (error) {
+         console.log(error);
+     }
+   }
+},
+```
+
+We can display data in our vue file. To do that, some steps need to be taken:
+
+1. Import `mapState` from vuex, this is to help generate computed getter functions for us.
+
+```JavaScript
+<script>
+import { mapState } from 'vuex';
+```
+
+2. Call the API on a lifecycle hook `mounted` and employ the `dispatch` method to call the action.
+
+```JavaScript
+mounted () {
+    this.$store.dispatch('loadPosts')
+},
+```
+
+3. Add mapState on your computed:
+
+```JavaScript
+computed: mapState([
+        'posts'
+  ]),
+</script>
+```
+
+4. Finally display data on your template.
+
+```JavaScript
+<template>
+     <div>
+         <ul v-for="post in posts">
+             <li>{{ post.title }}</li>
+         </ul>
+      </div>
+</template>
+```
```

Comment:
Make sure you provide the whole code snippet either before or after. Also, getting the same errors as the previous snippet.